### ISSUE TITLE
create an "lcov" target in cmake (Trac #691)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#690]
There is currently a CMAKE_BUILD_TYPE for generating gcov based coverage binaries. Extend this to a make target that will "do the right thing" for generating [http://ltp.sourceforge.net/coverage/lcov.php lcov] reports. 

Migrated from https://code.icecube.wisc.edu/ticket/691
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:52:36", 
    "description": "There is currently a CMAKE_BUILD_TYPE for generating gcov based coverage binaries. Extend this to a make target that will \"do the right thing\" for generating [http://ltp.sourceforge.net/coverage/lcov.php lcov] reports. ", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547250756695945", 
    "component": "cmake", 
    "summary": "create an \"lcov\" target in cmake", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2012-11-01T18:02:54", 
    "milestone": "", 
    "owner": "nega", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20121102T13:39:45: 
* nega changed description from "" to "There is currently a CMAKE_BUILD_TYPE for generating gcov based coverage binaries. Extend this to a make target that will "do the right thing" for generating [http://ltp.sourceforge.net/coverage/lcov.php lcov] reports. "
* nega changed owner from "" to "nega"
* nega changed type from "defect" to "enhancement"
* nega changed status from "new" to "assigned"
* nega changed component from "booking" to "cmake"

**Comment by icecube-trac** [comment]
Trac update at 20141008T17:50:01: nega changed status from "assigned" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20150415T22:40:52: 
* nega commented:

> In changeset 2128:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2128"
> create a coverage target for use on the buildbots
> 
> build, instrument, test, report, and rsync into staging.
> closes #690
> ```


* nega changed resolution from "" to "fixed"
* nega changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:52:36: nwhitehorn commented:

> In changeset 170036:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170036"
> create a coverage target for use on the buildbots
> 
> build, instrument, test, report, and rsync into staging.
> closes #690
> ```
